src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -116,8 +116,12 @@ FilePath sessionLockFilePath(const FilePath& sessionDir)
    return sessionDir.complete("lock_file");
 }
 
-// session dir lock (initialized by attachToSourceDatabase)
-FileLock s_sessionDirLock;
+// session dir lock (lock is acquired within 'attachToSourceDatabase()')
+boost::shared_ptr<FileLock> createSessionDirLock()
+{
+   return FileLock::createDefault();
+}
+boost::shared_ptr<FileLock> s_pSessionDirLock = createSessionDirLock();
```

Comment:
Our uses of fork are fine (because they all call exec immediately) however the use of fork by the **multicore** package is not fine as it forks rsession and runs multiple copies of it in parallel, all of which can end at different times.

In short, let's not rely on the static destructor at all in favor of an explicit cleanup call at process exit.